src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutStorage.cs

```diff
@@ -189,6 +180,47 @@ static string ConvertDictionaryToString(ICollection data)
             return serializer.SerializeObject(data);
         }
 
+        public TimeoutData Peek(string timeoutId)
```

Comment:
As mentioned, I don't agree with this method name. Its an entity (get, delete) and not a message (peek/receive). Maybe an improvement for v6.